src/lua/alloc.c

```diff
@@ -0,0 +1,150 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2024, Tarantool AUTHORS, please see AUTHORS file.
+ */
+#include "lua/utils.h"
+
+#include <lauxlib.h> /* luaL_error */
+
+#define LUA_MEMORY_DEFAULT (2ULL * 1024 * 1024 * 1024)
+
+/**
+ * The default allocator function which is wrapped into a new one
+ * with the Lua memory limit checker.
+ */
+static lua_Alloc orig_alloc;
+/**
+ * The maximum allowed LuaJIT memory limit.
+ */
+static size_t memory = LUA_MEMORY_DEFAULT;
```

Comment:
Minor: `memory_limit` sounds more clear to me (`memory` associates with a "currently allocated memory").
Feel free to ignore.

Reply:
Applied. You're right it's clearer.